kratos/sources/constitutive_law.cpp

```diff
@@ -697,7 +701,7 @@ void ConstitutiveLaw::InitializeMaterialResponseKirchhoff (Parameters& rValues)
  */
 void ConstitutiveLaw::InitializeMaterialResponseCauchy (Parameters& rValues)
 {
-    KRATOS_ERROR << "Calling virtual function for InitializeMaterialResponseCauchy" << std::endl;
+    KRATOS_ERROR_IF(RequiresInitializeMaterialResponse()) <<  "Calling virtual function for InitializeMaterialResponseCauchy. Please implement InitializeMaterialResponseCauchy or RequiresInitializeMaterialResponse in case this CL does not require it" << std::endl;

```

Comment:
```suggestion
    KRATOS_ERROR_IF(this->RequiresInitializeMaterialResponse()) <<  "Calling virtual function for InitializeMaterialResponseCauchy. Please implement InitializeMaterialResponseCauchy or RequiresInitializeMaterialResponse in case this CL does not require it" << std::endl;
```